postgres_base_block.py

```diff
@@ -1,14 +1,66 @@
 from nio.block.base import Block
-from nio.util.discovery import discoverable
-from nio.properties import VersionProperty
+from nio.command import command
+from nio.util.discovery import not_discoverable
+from nio.properties import VersionProperty, StringProperty, IntProperty
+from psycopg2 import connect
 
 
-@discoverable
-class Example(Block):
+@not_discoverable
+@command('Connection status', method='connection_status')
+@command('Reconnect', method='connect')
+@command('Disconnect', method='disconnect')
+class PostgresBase(Block):
+    """A block for communicating with an postgres database.
 
-    version = VersionProperty('0.1.0')
+    Properties:
+        host(str): hostname of the database to connect to
+        port(int): postgres port on the host to connect to
+        db_name(str): name of the database on the host
+        user_name(str): user name credential of the postgres server
+        password(str): password credential of the postgres server
+    """
 
-    def process_signals(self, signals):
-        for signal in signals:
-            pass
-        self.notify_signals(signals)
+    version = VersionProperty('1.0.0')
+    host = StringProperty(title="Host",
+                          default="[[POSTGRES_HOST]]")
+    port = IntProperty(title="Port",
+                       default="[[POSTGRES_PORT]]")
+    db_name = StringProperty(title="DB Name", allow_none=False)
+    user_name = StringProperty(title="User Name", default="test")
+    password = StringProperty(title="Password", allow_none=False)
```

Comment:
Providing no password is probably valid in the case that the database didn't have any auth